### PR TITLE
Implement PrivateNetworks API endpoint

### PR DESCRIPTION
**Pull request by norrland**

### CODE REVIEW

File: privatenetworks_test.go [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestPrivateNetworksCreate(t *testing.T) {
+	c := &mockClient{body: `{ "response": { "privatenetwork":
+		{ "id": "pn-123ab", "name": "mynetwork", "ipv6aggregate": "fe00:123::/48"}}}`}
```
Comment (brother) [inline privatenetworks_test.go:12]:
```suggestion
		{ "id": "pn-123ab", "name": "mynetwork", "ipv6aggregate": "2001:db8::/48"}}}`}
```

Use RFC3849 documentation network (as long as no validation is done client side but I think we don't have this here).

Same basic problem exists in multiple places ofc =)

File: privatenetworks_test.go [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+	assert.Equal(t, 50.0, net.Price, "Price is correct")
+	assert.Equal(t, 12.5, net.Discount, "Discount is correct")
+	assert.Equal(t, 37.5, net.Total, "Total price is correct")
+}
+
+func TestPrivateNetworkSegmentsCreate(t *testing.T) {
+	c := &mockClient{body: `{ "response": { "privatenetworksegment":
+		{ "id": "266979ab-1e05-4fbc-b9e0-577f31c0d2e9",
+		"name": "mysegment", "ipv6subnet": "fe00:123:0:/64",
+		"ipv4subnet": "10.0.0.0/24", "datacenter": "dc-fbg1",
```
Comment (brother) [inline privatenetworks_test.go:118]:
```suggestion
		"ipv4subnet": "192.168.0.0/24", "datacenter": "dc-fbg1",
```

Use any of the TEST-NET ranges - 192.168.0.0/16, 198.51.100.0/24, 203.0.113.0/24

File: doc_test.go [hunk 3]
```diff
@@ -412,6 +412,115 @@ func ExampleNetworkCircuitService_List() {
...
+
+func ExamplePrivateNetworkService_CreateSegment() {
+	client := glesys.NewClient("CL12345", "your-api-key", "my-application/0.0.1")
+
+	params := glesys.CreatePrivateNetworkSegmentParams{
+		PrivateNetworkID: "pn-123ab",
+		Datacenter:       "dc-fbg1",
+		Name:             "kvm-segment",
+		Platform:         "kvm",
+		IPv4Subnet:       "192.168.0.0/24",
```
Comment (brother) [inline doc_test.go:480]:
```suggestion
		IPv4Subnet:       "192.0.2.0/24",
```

File: privatenetworks_test.go [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+		"name": "mysegment", "ipv6subnet": "2001:db8:0::/64",
+		"ipv4subnet": "192.168.0.0/24", "datacenter": "dc-fbg1",
+		"platform": "kvm"}}}`}
+	n := PrivateNetworkService{client: c}
+
+	params := CreatePrivateNetworkSegmentParams{
+		Name:       "mysegment",
+		Platform:   "kvm",
+		Datacenter: "dc-fbg1",
+		IPv4Subnet: "192.168.0.0/24",
```
Comment (brother) [inline privatenetworks_test.go:126]:
```suggestion
		IPv4Subnet: "192.0.2.0/24",
```

File: privatenetworks_test.go [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+}
+
+func TestPrivateNetworkSegmentsList(t *testing.T) {
+	c := &mockClient{body: `
+		{ "response": { "privatenetworksegments": [{
+		"id": "fb34a19a-392a-43ec-ab3f-0c5b73ad1234",
+		"name": "mysegment",
+		"platform": "kvm",
+		"datacenter": "dc-fbg1",
+		"ipv4subnet": "192.168.0.0/24",
```
Comment (brother) [inline privatenetworks_test.go:156]:
```suggestion
		"ipv4subnet": "192.0.2.0/24",
```

File: privatenetworks_test.go [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+		"ipv6subnet": "2001:db8:0::/64"
+		}, {
+		"id": "fb34a19a-392a-43ec-ab3f-0c5b73ad5678",
+		"name": "othersegment",
+		"platform": "kvm",
+		"datacenter": "dc-fbg1",
+		"ipv4subnet": "192.168.0.0/24",
```
Comment (brother) [inline privatenetworks_test.go:163]:
```suggestion
		"ipv4subnet": "192.0.2.0/24",
```

**Review by hejhansson — APPROVED** [review]
👏

**Pull request merged**